tests/test_conf.py

```diff
@@ -89,3 +93,14 @@ def method(self):
         self.assertFalse(is_callable(Class))
         self.assertTrue(is_callable(c))
         self.assertTrue(is_callable(c.method))
+
+
+class SettingsObjectTestCase(TestCase):
+
+    def test_get_setting_deprecated(self):
+        with override_settings(FILTERS_TEST_123=True):
+            with self.assertWarns(DeprecationWarning):
+                DEPRECATED_SETTINGS.append('TEST_123')
+                DEFAULTS['TEST_123'] = True
```

Comment:
This change seems to outlast `test_get_setting_deprecated` runtime. Maybe `unittest.mock.patch` would help here.